.github/CONTRIBUTING.md

```diff
@@ -1,8 +1,8 @@
 # CONTRIBUTING
 
-We're using [Travis CI](https://travis-ci.com) as a continuous integration system.
+We're using [GitHub Actions](https://docs.github.com/en/actions) as a continuous integration system.
  
-For details, see [`.travis.yml`](../.travis.yml). 
+For details, see the workflows in [.github/workflows](../.github/workflows). 
```

Comment:
related to `GrumPHP` when I try to run it on my machine it's failing in `master` (before any code changes). 

Is this something you're aware of @drupol or am I missing some dependency? I already installed composer dependencies.

```
GrumPHP is sniffing your code!

Running tasks with priority 0!
==============================

Running task 1/14: license... ✔
Running task 2/14: composer_require_checker... ✘
Running task 3/14: composer... ✔
Running task 4/14: composer_normalize... ✔
Running task 5/14: yamllint... ✔
Running task 6/14: jsonlint... ✔
Running task 7/14: phplint... ✔
Running task 8/14: twigcs...
Running task 9/14: phpcsfixer... ✔
Running task 10/14: phpcs... ✔
Running task 11/14: phpstan... ✘
Running task 12/14: psalm... ✔
Running task 13/14: phpspec... ✘
Running task 14/14: infection... ✘

Reply:
Strange.

In order to help, could you run:

`php -v`

Then:

`./vendor/bin/infection run -vvv`

Reply:
I'm running on PHP 8 maybe that's why? 😅

```php -v
PHP 8.0.3 (cli) (built: Mar  4 2021 20:45:17) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.3, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.3, Copyright (c), by Zend Technologies
```

Infection fails because PHPSpec fails:

```
vendor/bin/phpspec run                                                                       1 ↵

      loophp\collection\Collection

  31  ✔ can append
  64  ✔ can apply
 182  ✔ can associate
 215  ! can asyncMap (109ms)
        exception [err:Error("Loop exceptionally stopped without resolving the promise")] has been thrown.
 232  ✔ can be constructed from a file
 242  ✔ can be constructed from a generator
 261  ✘ can be constructed from a stream
        expected exception of class "InvalidArgumentException", but got [obj:TypeError] with the
        message: "get_resource_type(): Argument #1 ($resource) must be of type resource, GdImage given"
 282  ✔ can be constructed from a string
 295  ✔ can be constructed from an iterable
 320  ✔ can be constructed from an iterator
 327  ✔ can be constructed from empty
 336  ✔ can be constructed with a closure
 383  ✔ can be constructed with an arrayObject
 391  ✔ can be instantiated with withClosure
 409  ✔ can be json encoded
 423  ✔ can be returned as an array
 429  ✔ can cache
 465  ✔ can chunk
 484  ✔ can coalesce
 503  ✔ can collapse
 531  ✔ can column
 568  ✔ can combinate
 620  ✔ can combine
 632  ✔ can compact
 645  ✔ can contains
 664  ✔ can convert use a string as parameter
 673  ✔ can count its items
 680  ✔ can current
 700  ✔ can cycle
 724  ✔ can diff
 735  ✔ can diffKeys
 748  ✔ can distinct
 757  ✔ can do the cartesian product
 776  ✔ can drop
 787  ✔ can dropWhile
string(6) "------"
array(3) {
  ["name"]=>
  string(5) "debug"
  ["key"]=>
  int(0)
  ["value"]=>
  string(1) "a"
}
array(3) {
  ["name"]=>
  string(5) "debug"
  ["key"]=>
  int(1)
  ["value"]=>
  string(1) "b"
}
string(6) "------"
array(3) {
  ["name"]=>
  string(5) "debug"
  ["key"]=>
  int(0)
  ["value"]=>
  string(1) "a"
}
array(3) {
  ["name"]=>
  string(5) "debug"
  ["key"]=>
  int(1)
  ["value"]=>
  string(1) "b"
}
string(6) "------"
 827  ✔ can dump
 908  ✔ can duplicate
 921  ✔ can every
 977  ✔ can explode
1026  ✔ can falsy
1041  ✔ can filter
1075  ✔ can flatten
1122  ✔ can flip
1135  ✔ can fold from the left
1149  ✔ can fold from the right
1163  ✔ can foldleft1
1178  ✔ can foldright1
1193  ✔ can forget
1200  ✔ can frequency
1227  ✔ can get
1238  ✔ can get an iterator
1247  ✔ can get current
1264  ✔ can get key
1281  ✔ can get the first item
1292  ✔ can get the last item
1320  ✔ can group
1348  ✔ can groupBy
1425  ✔ can has
1486  ✔ can head
1495  ✔ can if then else
1524  ✔ can implode
1535  ✔ can init
1562  ✔ can inits
1574  ✔ can intersect
1585  ✔ can intersectKeys
1606  ✔ can intersperse
1721  ✔ can key
1737  ✔ can keys
1744  ✔ can limit
1761  ✔ can lines
1780  ✔ can map
1809  ✔ can match
1837  ✔ can merge
1870  ✔ can normalize
1893  ✔ can nth
1904  ✔ can nullsy
1923  ✔ can pack
1938  ✔ can pad
1947  ✔ can pair
1990  ✔ can partition
2021  ✔ can permutate
2061  ✔ can pipe
2101  ✔ can pluck
2178  ✔ can prepend
2199  ✔ can random
2237  ✔ can reduction
2249  ✔ can reverse
2264  ✔ can rsample
2275  ✔ can scale
2291  ✔ can scanleft
2316  ✔ can scanleft1
2331  ✔ can scanright
2358  ✔ can scanright1
2383  ✔ can shuffle
2427  ✔ can since
2499  ✔ can slice
2514  ✔ can sort
2599  ✔ can span
2621  ✔ can split
2649  ✔ can squash
2667  ✔ can tail
2674  ✔ can tails
2688  ✔ can takeWhile
2726  ✔ can transpose
2777  ✔ can truthy
2796  ✔ can unfold
2841  ✔ can unlines
2862  ✔ can unpack
2899  ✔ can unpair
2918  ✔ can until
2955  ✔ can unwindow
3090  ✔ can unwords
3118  ✔ can unwrap
3138  ✔ can unzip
3156  ✔ can use range
3205  ✔ can use range with value 1
3215  ✔ can use times with a callback
3228  ✔ can use times without a callback
3246  ✔ can window
3381  ✔ can words
3410  ✔ can wrap
3425  ✔ can zip
3457  ✔ fix bug 57
3482  ✔ is initializable

      loophp\collection\Iterator\ArrayCacheIterator

  19  ✔ can cache an iterator of type generator
  49  ✔ is initializable

      loophp\collection\Iterator\IterableIterator

  17  ✔ can be constructed with a callable which is not a generator
  23  ✔ can get a key
  33  ✔ can rewind
  59  ✔ can use next
  73  ✔ is initializable

      loophp\collection\Iterator\PsrCacheIterator

  21  ✔ can cache data
  83  ✔ can get the inner iterator
  93  ✔ is initializable

      loophp\collection\Iterator\RandomIterator

  19  ✔ can build an iterator with a random seed
  70  ✔ can build an iterator without a random seed
  82  ✔ can get the innerIterator
  87  ✔ can rewind
 125  ✔ is initializable

      loophp\collection\Operation\Coalesce

  18  ✔ can coalesce
  49  ✔ is initializable

      loophp\collection\Operation\Current

  18  ✔ can get current
  33  ✔ is initializable

      loophp\collection\Operation\Limit

  18  ✔ can set a limit
  33  ✔ can set an offset
  48  ✔ does not change anything when no parameters are provided
  59  ✔ is initializable

----  failed examples

        loophp/collection/Collection
 261  ✘ can be constructed from a stream
        expected exception of class "InvalidArgumentException", but got [obj:TypeError] with the
        message: "get_resource_type(): Argument #1 ($resource) must be of type resource, GdImage given"

----  broken examples

        loophp/collection/Collection
 215  ! can asyncMap (109ms)
        exception [err:Error("Loop exceptionally stopped without resolving the promise")] has been thrown.


8 specs
147 examples (145 passed, 1 failed, 1 broken)
689ms
```

Reply:
ah yes,... it's not yet PHP 8 fully compatible :(

Reply:
Hmm I thought that with https://github.com/loophp/collection/issues/56 sorted it would be.

Do you mean it might not be usable in other projects using PHP 8, or just that internally it's not compatible?

Also, is there a list of issues or milestones to hit in order to reach compatibility?

Reply:
It will be compatible with PHP 8 in June because it will be my main focus during that month.

Reply:
Sounds good, as I'm looking to use this in a project using PHP 8.

Could I help address some of the issues?

Reply:
Of course ! PRs are very welcome and as you can see, I'm quite available :)

Reply:
Cool, I will start looking into the issues I experienced on my local